### DISCUSSION TITLE
Split CanvasItem's top_level functionality into two properties

### DISCUSSION THREAD

**Discussion by MyKoalaDev** [godotengine/godot-proposals#6086] [2D]
As of beta11, the property top_level enables drawing on top of other CanvasItems and also disables inheriting transformations. From what I could make of reading the source code, these are two completely separate operations which can (and should) be split into two different properties. 

This would be incredibly useful for me as my implementation of physics interpolation currently depends on this structure:
PhysicsBody2D
---PhysicsInterpolation2D (extends Node2D)
------Sprite2D, etc.

Because I'm interpolating the transform of the PhysicsInterpolation2D node, I want to disable transform propagation for that specific node and calculate its transform manually. I can't use a plain Node and calculate child transforms because I need the YSorting functionality (as an aside, YSorting—in its current state—is also severely detrimental to my mental health). The top_level property allows for ignoring transform propagation, but it also draws the children on top of everything (bad!!!).

Currently, I'm reading the parent transform every physics frame and applying an interpolated transform every process frame, but even then I don't have the peace of mind that the transform is properly interpolated at all times.

**Comment by Calinou** [comment]
Consider using the RemoteTransform node to achieve this without relying on `top_level`. It also allows only applying position/rotation/scale in a global manner.

**Reply by Kartopod** [reply]
Having to use RemoteTransforms gets really annoying quick if you have a lot of nodes, It would be nice if top level was split into two properties. What is the reason it does both those things at the moment? I cannot think of any reasons why that would be useful
